SignalServiceKit/src/Messages/OWSMessageDecrypter.m

```diff
@@ -669,8 +669,10 @@ - (void)processException:(NSException *)exception envelope:(SSKProtoEnvelope *)e
 
         OWSAssertDebug(errorMessage);
         if (errorMessage != nil) {
-            [errorMessage saveWithTransaction:transaction];
             [LKSessionManagementProtocol handleDecryptionError:errorMessage.errorType forHexEncodedPublicKey:envelope.source using:transaction];
+            if (![LKSessionManagementProtocol isErrorMessageBeforeRestoration:errorMessage]) {
```

Comment:
I wouldn't categorize this as session management. Also we already have `isFriendRequestFromBeforeRestoration(...)` which basically does the same thing. It'd be better to just modify that and move it to `SessionMetaProtocol`.